### ISSUE TITLE
Front-end registration updates for team invites

### ISSUE THREAD

**Issue by shaun-technovation** [Iridescent-CM/technovation-app#4284]
Background:
- #4283 will make some back-end changes to validate a team invite
- #4245 will make some changes to the API response to include the team invite info

The front-end flow should be updated so that if the team invite is valid, the corresponding registration type (student or mentor) should be enabled and allow the invited person to register.

**Comment by shaun-technovation** [comment]
For mentors who are invited by a student, the mentor profile type will be pre-selected (like an admin registration invite), but if a student is invited, nothing will be pre-selected, there isn't a way to guaranteed way (that I can think of) that will determine whether the person registering would be a parent or student; maybe if we add that (parent/student) as an option when students are sending out team invites, then we could do the pre-selecting.

**Comment by shaun-technovation** [comment]
Another note, when a student or mentor registers via a team invite, the team invite will only get updated if they register with the same email address as the email address the invite was sent to. This is the way admin registration invites used to work too. I was going to address this in my current PR (#4293), but it's already pretty big (over 1000+ lines of code added), and it's standalone enough, so I created this ticket for it: #4308.

**Comment by shaun-technovation** [comment]
The ability to use invites from students to join their team until submission close date is now on QA. The official date range of when they can be used is season start date until submission close date. Heroku config settings would need to be changed to test the date ranges (I think the submission close date is the only important one, I added a beginning date to prevent potential loopholes about using them during another season).

Side note, accepting the team invite as another student is not very intuitive, it's the existing functionality, nothing has changed with it, but maybe something we want to revisit.

Also, I made a bunch of changes in #4245, creating a new service that combines the admin registration settings, admin registration invites, and these new team invites, so if anything is off or needs changing regarding anything registration related, let me know!
